src/Lecture1.hs

```diff
@@ -104,7 +117,28 @@ and finds a sum of the numbers inside this string.
 
 The string contains only spaces and/or numbers.
 -}
-strSum str = error "TODO"
+
+strSum :: [Char] -> Int
+strSum str1 = sumListAccu (splitToInts str1) 0
+    where
+      splitAccu :: [Char] -> [Char] -> [[Char]] -> [[Char]]
+      splitAccu str strAccu strListAccu
+          | (null str) && (null strAccu) = strListAccu
+          | (null str) = strAccu : strListAccu
+          | (head str == ' ') && (null strAccu) = splitAccu (tail str) [] strListAccu
+          | (head str == ' ') = [strAccu] ++ (splitAccu (tail str) [] strListAccu)
+          | otherwise = splitAccu (tail str) (strAccu ++ [head str]) strListAccu
```

Comment:
That's a nice implementation of split by spaces! 👍🏻 
There's already a standard function `words` that does this for you, so you can use it instead 📚